### PR TITLE
Spanish Translation: Reset password Form and View

### PR DESCRIPTION
**Pull request by oruburos**
I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] is from a uniquely-named feature branch and has been rebased on top of the latest master. (If I was asked to make more changes, I have made sure to rebase onto master then too)
* [x] is descriptively named

This PR introduces the Spanish translation for Reset Password form and views. Touches both files, respective translation files and reduxFormUtils for validation messages.

### CODE REVIEW

File: client/modules/User/components/ResetPasswordForm.jsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ function ResetPasswordForm(props) {
       <Button
         type="submit"
         disabled={submitting || invalid || pristine || props.user.resetPasswordInitiate}
-      >Send Password Reset Email
+      >{props.t('ResetPasswordForm.Submit')}
```
Comment (andrewn) [inline client/modules/User/components/ResetPasswordForm.jsx:27]: Is there a reason why this line uses `props.t(...)` but lines 14 and 17 use `t.(...)`?

File: client/modules/User/pages/ResetPasswordView.jsx [hunk 2]
```diff
@@ -23,19 +24,18 @@ function ResetPasswordView(props) {
...
           <p className="reset-password__submitted">
-            Your password reset email should arrive shortly. If you don&apos;t see it, check
-            in your spam folder as sometimes it can end up there.
+            {props.t('ResetPasswordView.Submitted')}
           </p>
           <p className="form__navigation-options">
-            <Link className="form__login-button" to="/login">Log In</Link>
-            &nbsp;or&nbsp;
-            <Link className="form__signup-button" to="/signup">Sign Up</Link>
+            <Link className="form__login-button" to="/login">{props.t('LoginView.Login')}</Link>
```
Comment (andrewn) [inline client/modules/User/pages/ResetPasswordView.jsx:36]: This has the `LoginView` prefix even though it's in the `ResetPasswordView` component.

File: client/modules/User/components/ResetPasswordForm.jsx (hunk 1 above)
Comment (oruburos) [reply client/modules/User/components/ResetPasswordForm.jsx:27]: My bad, I forgot to update the last occurrence of props.t to t. I will update that

**Comment by oruburos** [comment]
@andrewn Hi, I move keys to ResetPasswordView and move the reference from props.t to t. Can you please have a look at this?

**Pull request merged**